### PR TITLE
feat(javascript rule): add dom purify lib

### PR DESCRIPTION
**Pull request by vjerci**
## Description
This pr adds support for dom purify vulnerability, as well as fixes santization library decteion for unsecure_html rule 

## Checklist

- [x] I've added test coverage that shows my fix or feature works as expected.
- [x] I've updated or added documentation if required.
- [x] I've included usage information in the description if CLI behavior was updated or added.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: pkg/commands/process/settings/rules/javascript/third_parties/dom_purify.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+severity:
+  default: low
+  PII: critical
+  PHI: medium
+  PD: high
+metadata:
+  description: "Unsecure use of DOMPurify detected."
+  remediation_message: |
+    ## Description
+    There are xss voulnerabilites when using DOMPurify's sanitize without proper configuration
```
Comment (elsapet) [inline pkg/commands/process/settings/rules/javascript/third_parties/dom_purify.yml]:
```suggestion
    There are xss vulnerabilites when using DOMPurify's sanitize without proper configuration
```

File: pkg/commands/process/settings/rules/javascript/third_parties/dom_purify.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    ```
+
+    or: 
+
+    ```
+      DOMPurify.sanitize(htmlToSanitize, { RETURN_DOM_FRAGMENT: true })
+    ```
+
+    ## Resources
+    - [Voulnerability explained](https://research.securitum.com/mutation-xss-via-mathml-mutation-dompurify-2-0-17-bypass)
```
Comment (elsapet) [inline pkg/commands/process/settings/rules/javascript/third_parties/dom_purify.yml]:
```suggestion
    - [Vulnerability explained](https://research.securitum.com/mutation-xss-via-mathml-mutation-dompurify-2-0-17-bypass)
```

Comment (cfabianski) [inline pkg/commands/process/settings/rules/javascript/third_parties/dom_purify.yml]:
```suggestion
    - [Vulnerability explained](https://research.securitum.com/mutation-xss-via-mathml-mutation-dompurify-2-0-17-bypass)
```

File: pkg/commands/process/settings/rules/javascript/third_parties/dom_purify.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  cwe_id:
+    - 79
+  associated_recipe: Sentry
```
Comment (elsapet) [inline pkg/commands/process/settings/rules/javascript/third_parties/dom_purify.yml]: Should this be here? 

```suggestion
```

File: pkg/commands/process/settings/rules/javascript/third_parties/dom_purify.yml (hunk 1 above)
Comment (cfabianski) [inline pkg/commands/process/settings/rules/javascript/third_parties/dom_purify.yml]:
```suggestion
    There are XSS vulnerabilites when using DOMPurify's sanitize without proper configuration.
```

**Review by elsapet — APPROVED** [review]

File: pkg/commands/process/settings/rules/javascript/third_parties/dom_purify.yml (hunk 3 above)
Comment (vjerci) [reply pkg/commands/process/settings/rules/javascript/third_parties/dom_purify.yml]: nop, it is there by accident, thank you

File: pkg/commands/process/settings/rules/javascript/lang/dangerous_insert_html.yml [hunk 4]
```diff
@@ -41,8 +41,8 @@ auxiliary:
           - variable: STRING_LITERAL
             detection: string_literal
             contains: false
-      - sanitize($<_>)
-      - sanitizeHTML($<_>)
+      - $<_>.sanitize($<...>$<_>$<...>)
+      - $<_>.sanitizeHTML($<...>$<_>$<...>)
```
Comment (cfabianski) [inline pkg/commands/process/settings/rules/javascript/lang/dangerous_insert_html.yml]:
```suggestion
      - $<_>.sanitize()
      - $<_>.sanitizeHTML()
```

Why not simply this?

Comment (vjerci) [reply pkg/commands/process/settings/rules/javascript/lang/dangerous_insert_html.yml]: think it will match only empty function calls?

Comment (vjerci) [reply pkg/commands/process/settings/rules/javascript/lang/dangerous_insert_html.yml]: will test it

Comment (vjerci) [reply pkg/commands/process/settings/rules/javascript/lang/dangerous_insert_html.yml]: seems to work

**Pull request merged**